codonPython/nhsNumberGenerator.py

```diff
@@ -41,9 +41,14 @@ def nhsNumberGenerator(to_generate: int, random_state: int = None)->list:
         digits = [int(digit) for digit in str(number)]
         # Apply weighting to digits
         weighted_digits = [(10 - index) * digit for (index, digit) in enumerate(digits)]
```

Comment:
Simplify this further by using a dot product. eg. `np.dot(list(map(int, list(str(number)))), np.arange(10,1,-1) )` (Do not use this one liner, I have it just as an example)